node/src/reactor.rs

```diff
@@ -443,33 +470,76 @@ where
         self.event_count += 1;
     }
 
-    // Handles dumping queue contents to files in /tmp.
-    async fn dump_queues(&self) {
+    /// Gets both the allocated and total memory from sys-info + jemalloc
+    fn get_allocated_memory() -> Option<AllocatedMem> {
+        let mem_info = match sys_info::mem_info() {
+            Ok(mem_info) => mem_info,
+            Err(error) => {
+                error!(%error, "unable to get mem_info using sys-info");
+                return None;
+            }
+        };
+
+        // mem_info gives us kb
+        let total = mem_info.total * 1024;
+
+        match jemalloc_epoch::mib() {
+            Ok(mib) => {
+                // jemalloc_ctl requires you to advance the epoch to update its stats
+                if let Err(advance_error) = mib.advance() {
+                    error!(%advance_error, "unable to advance jemalloc epoch");
+                }
+            }
+            Err(error) => {
+                error!(%error, "unable to get epoch::mib from jemalloc");
+                return None;
+            }
+        }
+        let allocated = match jemalloc_allocated::mib() {
+            Ok(allocated_mib) => match allocated_mib.read() {
+                Ok(value) => value as u64,
+                Err(error) => {
+                    error!(%error, "unable to read allocated mib using jemalloc");
+                    return None;
+                }
+            },
+            Err(error) => {
+                error!(%error, "unable to get allocated mib using jemalloc");
+                return None;
+            }
+        };
+
+        Some(AllocatedMem { allocated, total })
+    }
+
+    /// Handles dumping queue contents to files in /tmp.
+    async fn dump_queues(&mut self) {
         let timestamp = Timestamp::now();
+        self.last_queue_dump = Some(timestamp);
         let output_fn = format!("/tmp/queue_dump-{}.json", timestamp);
         let mut serializer = serde_json::Serializer::pretty(match File::create(&output_fn) {
             Ok(file) => file,
             Err(error) => {
-                tracing::error!(%error, "could not create output file ({}) for queue snapshot", output_fn);
+                error!(%error, "could not create output file ({}) for queue snapshot", output_fn);
                 return;
             }
         });
 
         if let Err(error) = self.scheduler.snapshot(&mut serializer).await {
-            tracing::error!(%error, "could not serialize snapshot to {}", output_fn);
+            error!(%error, "could not serialize snapshot to {}", output_fn);
```

Comment:
This _could_ be an error, given it is not likely to ever work if it failed once, but even here I'd consider a warning.